samples/docdb-scaling/scaleupdown.py

```diff
@@ -0,0 +1,204 @@
+            num_pending = num_pending + 1
+        if instance_status == 'failed':
+                num_failed = num_failed + 1
+    istatus["num_instances"]=len(instances)
+    istatus["num_available"]=num_available
+    istatus["num_pending"]=num_pending
+    istatus["num_failed"]=num_failed
+    return istatus
+    
+def getClusterInstances(cluster_id):
+    r = docdb.describe_db_clusters( DBClusterIdentifier=cluster_id)
+    cluster_info = r['DBClusters'][0]
+    if cluster_info['Status'] != 'available':
+        return False
+    else:
+        existing_instances = {}
+        for member in cluster_info['DBClusterMembers']:
+            member_id = member['DBInstanceIdentifier']
+            if member['IsClusterWriter']==True:
+                existing_instances[member_id]='WRITER'
+            else:
+                existing_instances[member_id]='READER'
+    
+    return existing_instances
+
+def getWriterInstance(cluster_id):
+    instances=getClusterInstances(cluster_id=cluster_id)
+    reversed_dict = {}
+    for key, value in instances.items():
+        reversed_dict.setdefault(value, [])
+        reversed_dict[value].append(key)
+    return reversed_dict["WRITER"][0]
+
+def isClusterHealthy(cluster_id):
+    istatus=getInstanceStatus(cluster_id)
+    LOGGER.info("cluster instance status: " + str(istatus))
+    if istatus==None:
+        LOGGER.error("Make sure cluster is in started state")
+        return False
+    if istatus["num_available"]==istatus["num_instances"] and isClusterStatusAvailable(cluster_id) :
+        return True
+    return False
+    
+def addInstances(cluster_id,desired_size,count):
+    ninstances=[]
+    for idx in range(count):
+        instance_iden=cluster_id + '-' + str(idx) + '-' + str(int(time.time()))
+        docdb.create_db_instance(DBInstanceIdentifier=instance_iden,DBInstanceClass=desired_size,Engine="docdb",DBClusterIdentifier=cluster_id)
+        ninstances.append(instance_iden)
+    return ninstances
+        
+
+def deleteInstances(instanceIdList):
+    for instance in instanceIdList:
+        docdb.delete_db_instance(DBInstanceIdentifier=instance)
+        
+def perfromFailover(cluster_id,failover_target):
+    docdb.failover_db_cluster(DBClusterIdentifier=cluster_id,TargetDBInstanceIdentifier=failover_target)
+    
+def checkExistingInstanceClass(cluster_id,itarget):
+    r = docdb.describe_db_instances(Filters=[{'Name':'db-cluster-id','Values': [cluster_id]}])
+    instances = r['DBInstances']
+    iclist=[]
+    for i in instances:
+        iclist.append(i['DBInstanceClass'])
+    if itarget in iclist:
+        return True
+    else:
+        return False
+
+def checkClusterHealth(cluster_id,mins=15):
+    count=0
+    while not isClusterHealthy(cluster_id=cluster_id):
+        count=count+1
+        if count==mins:
+            LOGGER.error("Cluster status is not healthy after " + str(mins) +  " mins. Please check the cluster health .Exiting.. ")
```

Comment:
What is our guidance here? Needs to be more than "Please check the cluster health".

Reply:
They need to make sure cluster is in healthy state before running script. Added details.